### ISSUE TITLE
Audio books playback in Android Auto support

### ISSUE THREAD

**Issue by jcdick1** [jellyfin/jellyfin-android#247]
**Describe the feature you'd like**
Add the Books library type to the Android Auto support, to allow playback of audio books

**Comment by nielsvanvelzen** [comment]
We need to add the "Books" type here:
https://github.com/jellyfin/jellyfin-android/blob/master/app/src/main/java/org/jellyfin/mobile/media/car/LibraryBrowser.kt#L187

I'm not sure if that's enough to get this functionality working so some testing needs to be done for that.

**Comment by jcdick1** [comment]
No real hurry, I just wanted to get it on a list somewhere for when there's time.  

My next question to go with this would be to know if automatically playing the next track is "pushed" as a server-side function or "pulled" as a client side, since all of my audio content is episodic, or file-per-chapter type, rather than one file per title.  So I know where to put in a feature request for that support.

**Comment by nielsvanvelzen** [comment]
Playlists are partially clientside and partially serverside. There is a feature request for proper audiobook here: https://features.jellyfin.org/posts/243/audiobook-support

**Comment by jcdick1** [comment]
Oh, I didn't realize that when the client plays the next episode of a TV series, it was basically building a playlist in the background.  I though either the server was telling the client "Hey, there's another episode in this series" or the client was asking "Is there another episode?  Cool, I'll queue it up."

That's basically all I'm after for my audio, since it is basically episodic.  Show->season->episode just without moving pictures.

**Comment by Kitof** [comment]
I'm very interested in this enhancement too. Any news ?

**Comment by teacupx** [comment]
I tried to do the simple change of this line: https://github.com/jellyfin/jellyfin-android/blob/43a734f281b398eeb02680b92d3e225fffddb80c/app/src/main/java/org/jellyfin/mobile/media/car/LibraryBrowser.kt#L206
Changing it for:
`.filter { item -> (item.collectionType == CollectionType.Music) || (item.collectionType == CollectionType.Books) }`

That was enough to make the libraries marked as "Books" to show up in Android Auto. But browsing them was a mess, since the app expected them to be browsable by tags, instead of folders like "Books" collections are organized. As a result, some folders appeared under "Artists", but with no files inside of them.

So this change would probably need a specific browsing menu for "Books" collection.

**Comment by fireXtract** [comment]
Is there interest in this feature still? I can pick it up if nobody else is looking at it.

**Comment by teacupx** [comment]
> Is there interest in this feature still? I can pick it up if nobody else is looking at it.

Definitely, I am very interested in this feature. Right now, I am using screen mirroring apps in order to be able to listen to my audiobooks in my car (which require rooted phone and are more difficult to use than the Android Auto player)

**Comment by resi23** [comment]
I think Android TV has the same problem. I don't see the books in this category at all. I would like to play audio books.

**Comment by fireXtract** [comment]
Ok I have done digging, and I can get audiobooks to render and play on android auto, but its a very hostile user experience. The web experience isn't much better -- the existing flow every time you want to listen is Audiobooks > Author > Book > scroll until you see a chapter marked as not played, unless you're fortunate enough that "Continue Listening" populates reliably.

I _think_ I need to design a new API for this, and will prioritize utilizing it in the android app first.

I don't want to give the impression I'm begging the maintainers to do this, I'm simply organizing my thoughts and requesting feedback.

Gaps compared to alternate software phrased as features that I plan to work on:
- Clicking a book starts it from the beginning, or resumes from where you left off (who cares about the individual files/chapters)
  - If clicking a chapter is possible "Play from here" should be the default.
- When android auto starts, it automatically resumes your most recent audio, be it book or audiobook
  - We would probably need two skip buttons on either side of play, I hope the MediaCompat for android auto supports this, one for back 10 seconds one for back a chapter, likewise for forwards.
- The android auto progress bar should display the length of the book (hours), and text can represent the individual chapters.
  - Audiobooks with a file per chapter will need a layer of abstraction probably
- A download option should be available for an entire book, I shouldn't need to click download for every file/chapter.

**Comment by fireXtract** [comment]
Haven't been satsified with my results yet, so no PR. But I do have a workaround for "Continue Listening" not working for audiobooks.

- Dashboard > Playback > Resume 
  - Set both of the following to one minute
    - Minimum Audiobook resume in minutes: 1
    - Audiobook remaining minutes to resume: 1
    
I think the issues arise because we expect the whole book to be one file by default, so chapter split audiobooks don't behave nicely. Another nifty workaround is to just merge the files into one per book.

Another log in my above list for later reference:
- Unfortunately "Play all from here" resets progress on a chapter
- Continue listening does not currently queue up the following chapters in a book

**Comment by jordanmichaelrushing** [comment]
@fireXtract how's progress going on this? Do you have a PR anywhere so others can look at, test, contribute, etc? I'm on an audiobook kick right now and there's a lot I want to fix with them on Jellyfin in app, but not being able to select at all in Android Auto is killing me.

@cameocoder i see you referencing this issue in some PR that i can't see. Any progress to share here?
